portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -26,30 +32,49 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private final List<String> comments = new ArrayList<>();
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    List<String> comments = new ArrayList<>();
+    comments = readFromDatastore(comments);
+
```

Comment:
You can do the new ArrayList<>() inside the readFromDatastore(); That way you don't need to create it outside and pass it in to the method.
The doGet can be
List<String> comments = readFromDatastore();